lib/std/io.zig

```diff
@@ -553,10 +561,15 @@ pub fn Poller(comptime StreamEnum: type) type {
                     self.windows.active.count,
                     &self.windows.active.handles_buf,
                     0,
-                    os.windows.INFINITE,
+                    if (nanoseconds) |ns|
+                        @min(std.math.cast(u32, ns / std.time.ns_per_ms) orelse (os.windows.INFINITE - 1), os.windows.INFINITE - 1)
```

Comment:
This prevents a user from accidentally waiting forever when they could just want to wait for a really long time. Technically this same logic should also be present in `std.time.sleep` but I'll leave that for another PR.